cmd/clusterctl/client/cluster/installer.go

```diff
@@ -47,6 +57,12 @@ type ProviderInstaller interface {
 	Images() []string
 }
 
+// InstallOptions defines the options used to configure installation.
+type InstallOptions struct {
```

Comment:
There are 2 ways we could have done this:
**Option 1:**
 Define `InstallOptions`, as done in this PR, and pass that as an argument to the installer's `Install()` function.
 
 **Option 2:**
Add `waitProviders` and `waitProvidersTimeout` to the `providerInstaller` struct and add functions to set these properties.  This would mean we will have to modify the `ProviderInstaller` interface to add a new function that would look like `SetProviderWait(WaitOptions)`.

We will have to modify the interface in both the cases. I went with option 1 in this PR (smaller change to the interface). Looking for feedback on preference of option to go with. 
Thanks.

Reply:
I think Option 1 is fine. Seems to also be a frequently used pattern around here. It also makes it easier to add new options in the future.

Reply:
Not sure if the following option/suggestion is feasible
- Keep Install function as per the original interface 
- Define new InstallWithOptions that takes the install option struct there by reducing breaking change impact
And eventually one can deprecate out Install() if need be

Reply:
This should work. But I think given https://github.com/kubernetes-sigs/cluster-api/pull/4934/files#r683272400 we don't want to invest the effort and go with the current implementation.